test/byContract/registries/registries.test.ts

```diff
@@ -0,0 +1,330 @@
+
+import { BytesLike } from '@ethersproject/bytes';
+import { AddressZero } from '@ethersproject/constants';
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import hre, { deployments, ethers, expect } from 'hardhat';
+import { loadHardhatFixtureDeployment } from '../../../scripts/hardhat-fixture-deployer';
+import {
+    AppRegistry,
+    AppRegistry__factory,
+    DatasetRegistry,
+    DatasetRegistry__factory,
+    IexecInterfaceNative,
+    IexecInterfaceNative__factory,
+    WorkerpoolRegistry,
+    WorkerpoolRegistry__factory,
+} from '../../../typechain';
+import { getIexecAccounts } from '../../../utils/poco-tools';
+const constants = require('../../../utils/constants');
+const randomAddress = () => ethers.Wallet.createRandom().address;
+
+describe('Registries', () => {
+    let proxyAddress: string;
+    let [iexecPoco, iexecPocoAsAdmin]: IexecInterfaceNative[] = [];
+    let [iexecAdmin, appProvider, datasetProvider, scheduler, anyone]: SignerWithAddress[] = [];
+
+    let ensRegistryAddress: string;
+    let appRegistry: AppRegistry;
+    let datasetRegistry: DatasetRegistry;
+    let workerpoolRegistry: WorkerpoolRegistry;
+
+    beforeEach(async () => {
+        proxyAddress = await loadHardhatFixtureDeployment();
+        await loadFixture(initFixture);
+    });
+
+    async function initFixture() {
+        ({ iexecAdmin, appProvider, datasetProvider, scheduler, anyone } =
+            await getIexecAccounts());
+        ensRegistryAddress = (await deployments.get('ENSRegistry')).address;
+        const appRegistryAddress = (await deployments.get('AppRegistry')).address;
+        appRegistry = AppRegistry__factory.connect(appRegistryAddress, iexecAdmin);
+        const datasetRegistryAddress = (await deployments.get('DatasetRegistry')).address;
+        datasetRegistry = DatasetRegistry__factory.connect(datasetRegistryAddress, iexecAdmin);
+        const workerpoolRegistryAddress = (await deployments.get('WorkerpoolRegistry')).address;
+        workerpoolRegistry = WorkerpoolRegistry__factory.connect(
+            workerpoolRegistryAddress,
+            iexecAdmin,
+        );
+
+        iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, anyone);
+        iexecPocoAsAdmin = iexecPoco.connect(iexecAdmin);
+    }
+
+    describe('Registry', () => {
+        it('Should retrieve base URI', async () => {
+            const chainId = hre.network.config.chainId;
+            expect(await appRegistry.baseURI()).to.equal(
+                `https://nfts-metadata.iex.ec/app/${chainId}/`,
+            );
+            expect(await datasetRegistry.baseURI()).to.equal(
+                `https://nfts-metadata.iex.ec/dataset/${chainId}/`,
+            );
+            expect(await workerpoolRegistry.baseURI()).to.equal(
+                `https://nfts-metadata.iex.ec/workerpool/${chainId}/`,
+            );
+        });
```

Comment:
Hard to add should set name ?

Reply:
Done [here](https://github.com/iExecBlockchainComputing/PoCo/pull/122/commits/a180f4b057dfdc54d2bb3c8187c23823bfc2b685)